### PR TITLE
nodejs-week3

### PR DESCRIPTION
**Pull request by oleksandra-lavryk**

### CODE REVIEW

File: nodejs/week3/warmup/api/calculator-router.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+function makeCalculation(requestQuery, response, action, initialValue) {
+  if (Object.keys(requestQuery).length === 0) {
+    response.send({ Message: "No parameters" });
+    return;
+  }
+  let result;
+  switch (action) {
+    case "+":
+      result = Object.values(requestQuery).reduce(
```
Comment (jslambda) [inline nodejs/week3/warmup/api/calculator-router.js:50]: Good idea to use the reduce function for this problem 👍

File: nodejs/week3/warmup/api/calculator-router.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        (previousValue, currentValue) => {
+          if (
+            previousValue &&
+            currentValue &&
+            previousValue != 0 &&
+            currentValue != 0 &&
+            !isNaN(Number(previousValue)) &&
+            !isNaN(Number(currentValue))
+          ) {
+            return Number(previousValue) / Number(currentValue);
```
Comment (jslambda) [inline nodejs/week3/warmup/api/calculator-router.js:89]: Division by zero is not valid. Can you also handle that?

File: nodejs/week3/warmup/api/calculator-router.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            currentValue &&
+            !isNaN(Number(previousValue)) &&
+            !isNaN(Number(currentValue))
+          ) {
+            return Number(previousValue) - Number(currentValue);
+          }
+        }
+      );
+      break;
+    case "/":
```
Comment (jslambda) [inline nodejs/week3/warmup/api/calculator-router.js:78]: What would happen if there are fewer than two query parameters in case of division?

Comment (oleksandra-lavryk) [reply nodejs/week3/warmup/api/calculator-router.js:78]: @jslambda  I fixed that part. Thanks!

File: nodejs/week3/warmup/api/calculator-router.js (hunk 2 above)
Comment (oleksandra-lavryk) [reply nodejs/week3/warmup/api/calculator-router.js:89]: @jslambda  Fixed that. Thanks.

**Pull request merged**